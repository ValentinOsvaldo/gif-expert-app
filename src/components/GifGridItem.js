import React from "react";

const GifGridItem = ({ id, title, url }) => {
<<<<<<< HEAD
    return (
        <figure className="card">
            <img src={ url } alt={ title } />
            <figcaption>
                { title }
            </figcaption>
        </figure>
    )
}

export default GifGridItem
=======
  return (
    <figure className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <figcaption className="card-name">
        <p>{title}</p>
      </figcaption>
    </figure>
  );
};
>>>>>>> c04cb987e99ecbf7c85add35aa47555362b6222b

export default GifGridItem;
