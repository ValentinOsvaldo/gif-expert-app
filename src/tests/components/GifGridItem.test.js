import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("<GifGridItem />", () => {
  const title = "Title",
    url = "https://localhost/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);
  });

  test("Debe tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe tener animate__fadeIn", () => {
    const figure = wrapper.find("figure");
    const className = figure.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
