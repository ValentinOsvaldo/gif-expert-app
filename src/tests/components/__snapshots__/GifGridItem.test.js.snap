// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> Mostrar correctamente el componente 1`] = `
<figure
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="https://localhost/algo.jpg"
  />
  <figcaption
    className="card-name"
  >
    <p>
      Title
    </p>
  </figcaption>
</figure>
`;
